sdc/datatypes/hpat_pandas_dataframe_rolling_functions.py

```diff
@@ -0,0 +1,141 @@
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+import pandas
+
+from sdc.datatypes.common_functions import TypeChecker
+from sdc.datatypes.hpat_pandas_dataframe_rolling_types import DataFrameRollingType
+from sdc.hiframes.pd_dataframe_ext import get_dataframe_data
+from sdc.utils import sdc_overload_method
+
+
+sdc_pandas_dataframe_rolling_docstring_tmpl = """
+    Intel Scalable Dataframe Compiler User Guide
+    ********************************************
+    Pandas API: pandas.core.window.Rolling.{method_name}
+{limitations_block}
+    Examples
+    --------
+    .. literalinclude:: ../../../examples/dataframe/rolling/dataframe_rolling_{method_name}.py
+       :language: python
+       :lines: 27-
+       :caption: {example_caption}
+       :name: ex_dataframe_rolling_{method_name}
+
+    .. command-output:: python ./dataframe/rolling/dataframe_rolling_{method_name}.py
+       :cwd: ../../../examples
+
+    .. seealso::
+        :ref:`DataFrame.rolling <pandas.DataFrame.rolling>`
+            Calling object with a DataFrame.
+        :ref:`DataFrame.rolling <pandas.DataFrame.rolling>`
+            Calling object with a DataFrame.
+        :ref:`DataFrame.{method_name} <pandas.DataFrame.{method_name}>`
+            Similar method for DataFrame.
+        :ref:`DataFrame.{method_name} <pandas.DataFrame.{method_name}>`
+            Similar method for DataFrame.
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+
+    Pandas DataFrame method :meth:`pandas.DataFrame.rolling.{method_name}()` implementation.
+
+    .. only:: developer
+
+    Test: python -m sdc.runtests -k sdc.tests.test_rolling.TestRolling.test_dataframe_rolling_{method_name}
+
+    Parameters
+    ----------
+    self: :class:`pandas.DataFrame.rolling`
+        input arg{extra_params}
+
+    Returns
+    -------
+    :obj:`pandas.DataFrame`
+         returns :obj:`pandas.DataFrame` object
+"""
+
+
+def params2list(params):
+    """Convert parameters from dict to list"""
+    return ['{}={}'.format(k, v) for k, v in params.items()]
+
+
+def sdc_pandas_dataframe_rolling_apply(method_name, self, args=None, kws=None):
```

Comment:
Could you please separate code generation and receiving python object part?

So we would have function which returns generated function code as text and global vars needed to it?